README.md

```diff
@@ -27,6 +29,46 @@ If you provide a zap.Error tag, the related stacktrace will also be attached.
 
 Additionally there is a tracer(opentracing/jaeger) available in the logger which should be closed before exiting main.
 
+### Problems
+
+`problems` is a small library which implements the RFC7807 error response format standard for e.g. HTTP API's.
+
+The `problems` lib provides a struct called `Problem` and an interface called `ProblemInfo`.
+`Problem` implements the `error` interface, so you can simply return the problem as an error in your application.
+
+```go
+func returnError() error {
+    return problems.New()
+}
+```
+
+You can marshal the `Problem` struct to JSON:
+
+```go
+func makeJSON(problem *Problem) ([]byte, error) {
+    return json.Marshal(&problem)
+}
+```
+
+If you want to define a HTTP problem, you can do it like this:
+
+```go
+var problemNotFound = problems.New().SetTitle("NotFound").SetDetail("The requested url was not found.").SetStatus(404).SetType("https://example.com/problem/description")
+```
+
+For custom fields for more detailed problem information, you can implement `Problem` in your custom struct:
+
+```go
+type customInfo struct {
+    AdditionalField1 bool `json:"additional_field_1"`
+}
+
+type customProblem struct {
+    *problems.Problem
```

Comment:
problems.Problem does not match the package errors.

Also the name Problem does not convince me too much.